### PR TITLE
docs: add docs for key-generate

### PR DESCRIPTION
**Pull request by riyazdf**
Signed-off-by: Riyaz Faizullabhoy <riyaz.faizullabhoy@docker.com>

### CODE REVIEW

File: docs/reference/commandline/trust_key_generate.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+     another repository, consider it read-only there, as it will
+     periodically be overwritten by the definitive file. Pull
+     requests which include edits to this file in other repositories
+     will be rejected.
+-->
+
+# trust key-generate
+
+```markdown
+Usage:  docker trust key-generate [OPTIONS] NAME
```
Comment (eiais) [inline docs/reference/commandline/trust_key_generate.md]: I don't think key generate has options this should be fixed in key_generate.go as well

Comment (riyazdf) [reply docs/reference/commandline/trust_key_generate.md]: good catch.  I'll also have to update `sign` and `view` upstream.

File: cli/command/trust/sign.go [hunk 2]
```diff
@@ -21,7 +21,7 @@ import (
 
 func newSignCommand(dockerCli command.Cli) *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "sign [OPTIONS] IMAGE:TAG",
+		Use:   "sign IMAGE:TAG",
```
Comment (eiais) [inline cli/command/trust/sign.go:24]: sign is about to have the --local flag, is it worth removing it just to add it right back?

**Review by eiais — APPROVED** [review]

File: cli/command/trust/sign.go (hunk 2 above)
Comment (riyazdf) [reply cli/command/trust/sign.go:24]: let's add it in the `--local` flag PR explicitly.  That PR could even go against upstream master once we're ready

**Pull request merged**